commit d8dc7133c2c8255f194e1fa5e550ef1bd4796bf6
Author: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>
Date:   Thu Oct 15 10:12:24 2020 +0200

    mvapich2 remove deprecated sys_siglist
    
    sys_siglist has been deprecated and even removed from newer glibc version.
    Use strsignal as a replacement
    
    Signed-off-by: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>

diff --git src/mpid/ch3/channels/common/src/util/error_handling.c src/mpid/ch3/channels/common/src/util/error_handling.c
index b6c67c41c056..27f7e63a64ac 100644
--- src/mpid/ch3/channels/common/src/util/error_handling.c
+++ src/mpid/ch3/channels/common/src/util/error_handling.c
@@ -69,7 +69,7 @@ int show_backtrace = 0;
 // Signal handler for errors
 void error_sighandler(int sig, siginfo_t *info, void *secret) {
     // Always print error
-    PRINT_ERROR( "Caught error: %s (signal %d)\n", sys_siglist[sig], sig );
+    PRINT_ERROR( "Caught error: %s (signal %d)\n", strsignal(sig), sig );
     // Show backtrace if required
     if (show_backtrace) print_backtrace();
     // Raise the signal again with default handler
diff --git src/pm/mpirun/mpispawn.c src/pm/mpirun/mpispawn.c
index dd0c6f06f1c5..bcb562ea7552 100644
--- src/pm/mpirun/mpispawn.c
+++ src/pm/mpirun/mpispawn.c
@@ -800,7 +800,7 @@ void child_handler(int signal)
     gethostname(my_host_name, MAX_HOST_LEN);
 
     rank = mt_id;
-    PRINT_DEBUG(DEBUG_Fork_verbose, "mpispawn child_handler: got signal %d: %s\n", signal, sys_siglist[signal]);
+    PRINT_DEBUG(DEBUG_Fork_verbose, "mpispawn child_handler: got signal %d: %s\n", signal, strsignal(signal));
     while (1) {
         do {
             pid = waitpid(-1, &status, WNOHANG);
